WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit External Link/ProductExternalLinkViewController.swift

```diff
@@ -0,0 +1,232 @@
+import UIKit
+import Yosemite
+
+/// Contains text fields for editing an external product's external URL and button text.
+final class ProductExternalLinkViewController: UIViewController {
+    private lazy var tableView: UITableView = {
+        return UITableView(frame: .zero, style: .grouped)
+    }()
+
+    private let product: Product
+    private let sections: [Section]
+
+    private var externalURL: String?
+    private var buttonText: String
+
+    private lazy var keyboardFrameObserver: KeyboardFrameObserver = {
+        let keyboardFrameObserver = KeyboardFrameObserver { [weak self] keyboardFrame in
+            self?.handleKeyboardFrameUpdate(keyboardFrame: keyboardFrame)
+        }
+        return keyboardFrameObserver
+    }()
```

Comment:
You could make this shorter by:

```suggestion
    private lazy var keyboardFrameObserver = KeyboardFrameObserver { [weak self] keyboardFrame in
        self?.handleKeyboardFrameUpdate(keyboardFrame: keyboardFrame)
    }

```

Reply:
nice! updated in https://github.com/woocommerce/woocommerce-ios/pull/2442/commits/8a9971de59fadf9b837a9e21d92a724e23627d93